crates/pallets/starknet/src/lib.rs

```diff
@@ -166,7 +163,10 @@ pub mod pallet {
     #[pallet::hooks]
     impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
         /// The block is being finalized.
-        fn on_finalize(_n: T::BlockNumber) {
+        fn on_finalize(n: T::BlockNumber) {
+            if UniqueSaturatedInto::<u64>::unique_saturated_into(n) >= 1 {
+                assert!(SeqAddrUpdate::<T>::take(), "Sequencer address must be set for the block");
+            }
```

Comment:
Rather than doing this check (if >= 1) in every block, when it's only false for the first one, can't we consider setting the storage to the default address in `GenesisBuild::build`.

This way it is set for the genesis block too, and we can remove this otherwise useless branching